website/static/js/registrationUtils.js

```diff
@@ -260,10 +260,15 @@ var MetaSchema = function(params) {
     self.fulfills = params.fulfills || [];
     self.messages = params.messages || {};
 
+    var count = 0;
     $.each(self.schema.pages, function(i, page) {
         var mapped = {};
         $.each(page.questions, function(qid, question) {
-            mapped[qid] = new Question(question, qid);
+            // This is necessary now since page.questions is a list
```

Comment:
This doesn't work because questionId needs to be unique for the whole schema, not just a page. `qid` in the second `$.each` will cause duplicates and the index for the first `$.each` will only count the number of pages so only a few questions will get ids. `count` is needed so it can be schema wide and also be incremented on a question basis